README.md

```diff
@@ -0,0 +1,335 @@
+| sample-ref        | `stix:sample_ref`       | `nfo:FileDataObject` | The file that was scanned                                         |
+|**TODO**result-name| `stix:result_name`      | `xsd:string`         | Details of the result, e.g. names of detected malware             |
+
+**TODO**: We could add `result-name`, but clamscan returns an array of
+strings, because more than 1 malware may be found in a file. We could
+write a JSON-string of that array to `result-name`, if that is not a
+problem.
+
+### Configuration
+
+#### Environment variables
+
+The following enviroment variables can be configured:
+
+* `LOG_INCOMING_DELTA (default: "false")`: Log the delta message as
+  received from the delta-notifier to the console.
+* `LOG_INCOMING_SCAN_REQUESTS (default: "false")`: Log the requests
+  received by endpoint `/scan`.
+* `VIRUS_SCANNER_CLAMD_USER (default: "clamav")`: User to run the
+  ClamAV daemon `clamd` as.
+* The environment variables recognized by
+  [mu-javascript-template](https://github.com/mu-semtech/mu-javascript-template/blob/v1.7.0/README.md#environment-variables).
+
+### REST API
+
+Notes:
+- In various error cases (e.g. no physical file IRI found, file not
+  found on disk, errors from clamscan), virus-scanner will create a
+  malware-analysis resource for the requested file IRI with `result`
+  `unknown`. Error details are logged. Those are not uncaught errors
+  that would lead to a 500 Server Error response.
+- Storing a malware-analysis in the database may fail with only a
+  logged error (e.g. if 202 Accepted was already returned), or even
+  silently (e.g. file IRI not in any graph at time of storing result).
+- [How to check if a file is clean](#how-to-check-if-a-file-is-clean)
+  describes how to ensure a file is clean considering such corner cases.
+
+#### POST /delta
+Accepts requests like those created by
+[delta-notifier](https://github.com/mu-semtech/delta-notifier).
+
+##### Response
+###### 202 Accepted
+Delta contains logical file IRI insertions that will be scanned.
+
+The results will be logged and stored in the database.
+
+###### 204 No Content
+Delta contains no logical file IRI insertions. No results are stored.
+
+###### 500 Server error
+Uncaught error.
+
+#### POST /scan
+Scan a file. Accepts a `multipart/form-data` with a `file` parameter
+containing a logical file IRI.
+
+##### Response
+###### 201 Created
+The newly created malware-analysis in the response body:
+
+```javascript
+{
+  "data": {
+    "type": "malware-analyses",
+    "id": "3a2cafd0-8f8a-11ee-a732-97ed1ab0131d",
+    "attributes": {
+      "uri": "http://data.gift/virus-scanner/analysis/id/3a2cafd0-8f8a-11ee-a732-97ed1ab0131d",
+      "analysis-started": "2023-11-30T14:10:33.855Z",
+      "analysis-ended": "2023-11-30T14:10:33.930Z",
+      "result": "malicious",
+      "sample-ref": "http://mu.semte.ch/services/file-service/files/65684a368d76fe0010000000"
```

Comment:
To be compatible with json:api, i think it should be releationships. (https://jsonapi.org//)